src/pytorch_lightning/core/datamodule.py

```diff
@@ -150,7 +150,7 @@ def predict_dataloader():
                 return [dataloader(ds) for ds in predict_dataset]
             return dataloader(predict_dataset)
 
-        datamodule = cls()
+        datamodule = cls(batch_size=batch_size)
```

Comment:
What do you mean which library? pytorch lightning library.

Reply:
I tried to use SemanticSegmentationData.from_dataset method

Reply:
and i get (sorry about pic of error, instead of str)
![image](https://user-images.githubusercontent.com/101559510/183401390-d9a5b70f-985b-43d5-8e03-ca881dff8665.png)

Reply:
From skimming the flash.DataModule source code, it doesn't look like this method is supposed to be compatible with it, as it already manages creating the DataLoaders etc.

https://github.com/Lightning-AI/lightning-flash/blob/644f2b559f87b40a5b623f5260eee0cf924ea0a5/flash/core/data/new_data_module.py

Thoughts @krshrimali?

Reply:
Flash has an extra parameter `**datamodule_kwargs` in their special `from_*` methods. But this is unfortunately not a universal solution, because a parameter like `batch_size`, which is also used as a parameter by the from_dataset method, can't be passed in "twice" as kwargs. There some solutions

1) use a proper dict datamodule_kwargs as parameter instead of kwargs. But this would be inconsistent with how Flash handles it.
2) Have datamodule_kwargs as proper kwargs, but do a signature inspection for the init to know whether you need to pass in parameters like batch_size
3) Override the from_datasets method in FlashDatamodule and raise an error that this method is not supported. 
4) Something else?

Reply:
i'd say `2`

Reply:
Implemented in #14185